app/lib/clients/vault/certificate.rb

```diff
@@ -38,36 +45,59 @@ def cert_engine_type
       end
 
       def enable_ca
-        # if mount exists, assume configuration is done
-        if client.sys.mounts.key?(intermediate_ca_mount.to_sym)
-          return false
-        end
-
-        # create the mount
+        # create the intermediate mount
         enable_engine(intermediate_ca_mount, cert_engine_type)
-        true
+      end
+
+      def configure_root_ca
+        return if client.sys.mounts.key?(root_ca_mount.to_sym)
+
+        # enable engine
+        enable_engine(root_ca_mount, cert_engine_type)
+
+        # generate root certificate
+        root_cert = client.logical.write("#{root_ca_mount}/root/generate/internal",
+                                        common_name: "astral.internal",
+                                        issuer_name: root_ca_ref,
+                                        ttl: "87600h").data[:certificate]
+        # save the root certificate
+        File.write("tmp/#{root_ca_mount}.crt", root_cert)
+
+        client.logical.write("#{root_ca_mount}/config/cluster",
+                            path: "#{vault_address}/v1/#{root_ca_mount}",
+                            aia_path: "#{vault_address}/v1/#{root_ca_mount}")
+
+        client.logical.write("#{root_ca_mount}/roles/2024-servers",
+                            allow_any_name: true,
```

Comment:
Should "2024-servers" be a define somewhere?

Reply:
good point -- I think i'll just remove this role for now (which addresses next comment as well).